modules/ocf_kubernetes/manifests/loadbalancer.pp

```diff
@@ -0,0 +1,38 @@
+class ocf_kubernetes::loadbalancer {
+  include ::haproxy
+
+  $kubernetes_workers_ipv4 = lookup('kubernetes::workers_ipv4')
+  $kubernetes_worker_nodes = lookup('kubernetes::worker_nodes')
+
+  haproxy::frontend { 'kubernetes-frontend':
+    mode    => 'http',
+    bind    => {
+      "${::ipaddress}:80"   => [],
+      # TODO: Listen on 443 when we add TLS
+      #"$::ipaddress:443"  =>  ['ssl', 'crt', '/path/to/cert.pem']
+    },
+    options => {
+      'default_backend'  => 'kubernetes-backend',
+      # TODO: enable once we add TLS
+      #'redirect' =>  'scheme https code 301 if !{ ssl_fc }',
+    };
+  } ->
+  haproxy::backend { 'kubernetes-backend':
+    options => {
+      option         => [
+        'forwardfor'
+      ],
+      'mode'         => 'http',
+      'balance'      => 'roundrobin',
+      'http-request' => 'set-header X-Forwarded-Port %[dst_port]',
+      # TODO: this does nothing until we add TLS
+      'http-request' => 'add-header X-Forwarded-Proto https if { ssl_fc }',
+    }
+  } ->
+  haproxy::balancermember { 'kubernetes-ingress-worker':
+    listening_service => 'kubernetes-frontend',
+    ports             => ['80'],
+    ipaddresses       => $kubernetes_workers_ipv4,
```

Comment:
I wonder if there's a way to do this without having to manually specify all the workers. Could we use Kubernetes internal IPs in combination with some sort of service discovery mechanism to know which ones are up?

Reply:
Not that I currently know of. A puppet `fact` could also work, but I don't see anything horribly wrong with having the worker IPs in an array.

Reply:
I do agree that updating the IPs each time is a pain, and even having a list of workers is a bit annoying, but we do that currently for the marathon/mesos setup, so I think it's probably fine to get it to a feature parity level at least.